xanalytics/__init__.py

```diff
@@ -0,0 +1,51 @@
+"""
+Functions for communicating with the xanalytics API.
+"""
+import logging
+import os
+from tempfile import mkstemp
+
+from django.conf import settings
+import requests
+
+
+log = logging.getLogger(__name__)
+
+
+def send_request(url, course_id):
+    """
+    Send initial request to xanalytics API.
+    Args:
+        url (string): URL of xanalytics API
+        course_id (int): Course primary key.
+    Returns:
+        token (unicode): Job token from xanalytics.
+    """
+    return requests.post(url=url, data={"course_id": course_id}).json()
+
+
+def get_result(url, token):
+    """
+    Get result from xanalytics for token.
+    Args:
+        token (unicode): Token from xanalytics job.
+    Returns:
+        data (dict): Dictionary from JSON object provided by xanalytics.
+    """
+    return requests.post(url=url, data={"token": token}).json()
+
+
+def get_xanalytics_file(url):
+    """
+    Retrieve file and save locally.
+    Args:
+        URL (unicode): URL to file location.
+    Returns:
+        filename (string): Path to file on local filesystem.
+    """
+    resp = requests.get(url)
```

Comment:
Why are we making this a file at all?  We delete it as soon as we process it, and the requests.get means it is already going to be completely stored in memory, so we aren't saving mem at all, seems to me we should just get rid of this function, do the request above and have the update function work off the in memory structure.